### DISCUSSION TITLE
Auto Gain Control ( AGC ) for Mic

### DISCUSSION THREAD

**Discussion by musiq021** [pschatzmann/arduino-audio-tools#1241] [Q&A] [answered]
Has anyone had any success in implementing some form of auto gain control for a microphone input? 

My i2s Mic ( ICS-43434 ) sounds fine at default values when speaking directly into it, but at any distance, requires considerable gain and subsequent clipping at close distance.

**Comment by pschatzmann** [comment, marked as answer]
It should be quite easy to build something which reads the current volume (using a VolumeOutput) and adjust it (using a [VolumeStream](https://github.com/pschatzmann/arduino-audio-tools/wiki/Volume-Control))
